compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -1402,6 +1402,10 @@ fn panic_on_forbidden_read<D: Deps>(data: &DepGraphData<D>, dep_node_index: DepN
 
     panic!(
         "Error: trying to record dependency on DepNode {dep_node} in a \
-         context that does not allow it (e.g. during query deserialization)."
+         context that does not allow it (e.g. during query deserialization). \
+         The most common case of recording a dependency on a DepNode `foo` is \
+         when the correspondng query `foo` is invoked. Invoking queries is not \
+         allowed as part of loading something from the incremental on-disk cache. \
+         See <https://github.com/rust-lang/rust/pull/91919>."
```

Comment:
That's a lot better, thanks!

Not sure if linking to the PR is very useful, though.

Reply:
I'll leave the link in unless you or @oli-obk think that it's actively hurting understandability.